odriveS1/src/odriveS1/__init__.py

```diff
@@ -2,8 +2,10 @@
 This files odriveS1 model with the Viam Registry.
 """
 
-from viam.resource.registry import Registry
+from viam.resource.registry import Registry, ResourceCreatorRegistration
 from viam.components.motor import Motor
 from .odriveS1 import OdriveS1
+from .odriveCAN import OdriveCAN
 
-Registry.register_resource_creator(Motor.SUBTYPE, OdriveS1.MODEL, OdriveS1.new)
+Registry.register_resource_creator(Motor.SUBTYPE, OdriveS1.MODEL, ResourceCreatorRegistration(OdriveS1.new))
+Registry.register_resource_creator(Motor.SUBTYPE, OdriveCAN.MODEL, ResourceCreatorRegistration(OdriveCAN.new))
```

Comment:
There has to be a better way to do this than adding both types to the registry. My first thought is to just register `Odrive.new` which should then actually register whichever class is specified in the config. I tried to do this and the resource could not be found in the registry, so I don't know if that has something to do with how I'm adding it here, or if it's a bug in `odrive.py`

Reply:
If you weren't to register a different model wouldn't you have to add a config value to determine what communication protocol the user wants?

Reply:
yes! that will be added to the config